routers/web/repo/repo.go

```diff
@@ -622,7 +622,20 @@ func SearchRepo(ctx *context.Context) {
 		}
 	}
 
-	var err error
+	// To improve performance when only the count is requested
+	if ctx.FormBool("count_only") {
+		count, err := repo_model.CountRepository(ctx, opts)
+		if err != nil {
+			ctx.JSON(http.StatusInternalServerError, api.SearchError{
+				OK:    false,
+				Error: err.Error(),
+			})
```

Comment:
I don't know from what time, the code started exposing internal error messages to end users. There were some worries about "the internal error message may contain sensitive information for server environment".


And, I don't see why it should use `api.SearchError` here, is the response really handled by frontend JS?

Reply:
Looks like it's a very old code. I sent a new commit 413b5c0 to return a simple error for non-admin users.

Reply:
> And, I don't see why it should use `api.SearchError` here, is the response really handled by frontend JS?

And, I don't see why it should use `api.SearchError` here, is the response really handled by frontend JS?




> Looks like it's a very old code. I sent a new commit [413b6c0](https://github.com/go-gitea/gitea/commit/413b6c0ebbc5796657ccca2f254c95b335286ca7) to return a simple error for non-admin users.

It doesn't look good to me if it would fill the code base with a lot of `if ctx.Doer != nil && ctx.Doer.IsAdmin {`

Reply:
> > And, I don't see why it should use `api.SearchError` here, is the response really handled by frontend JS?
> 
> And, I don't see why it should use `api.SearchError` here, is the response really handled by frontend JS?

I think this needs a refactor to adjust so I don't think we should do that in this PR because this PR needs to be backport to v1.21.

> > Looks like it's a very old code. I sent a new commit [413b6c0](https://github.com/go-gitea/gitea/commit/413b6c0ebbc5796657ccca2f254c95b335286ca7) to return a simple error for non-admin users.
> 
> It doesn't look good to me if it would fill the code base with a lot of `if ctx.Doer != nil && ctx.Doer.IsAdmin {`

Reply:
> I think this needs a refactor to adjust so I don't think we should do that in this PR because this PR needs to be backport to v1.21.

Maybe it doesn't need to be backported since it is a (trivial?) performance optimization, not a bug fix or necessary enhancement.

Reply:
Why not? It's not a big PR but will improve the dashboard performance.

Reply:
I do not think "because this PR needs to be backport to v1.21" is a strong reason to make the code more hacky.